### PR TITLE
ci: add sync to develop job for the release workflow

### PR DESCRIPTION
**Pull request by Sakchai-Refinitiv**
## Description
 Add the new job in the last step of release workflow to synchronize from `main` to `develop` branch automatically, for updating the changelogs and package version. We must have this feature to reduce human error by doing it manually.

## Type of change

- [x] CI Improvement

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -41,3 +41,10 @@ jobs:
         run: lerna publish from-git --yes
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Sync to develop branch
```
Comment (wsuwt) [inline .github/workflows/release.yml]:
```suggestion
      - name: Sync published changes to develop branch
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -41,3 +41,10 @@ jobs:
...
+        run: |
+          git checkout develop
+          git merge main --no-edit --no-ff
```
Comment (wsuwt) [inline .github/workflows/release.yml:48]: @Sarin-Udompanish  I wonder if commitlint would complain the merge commit here ?

Comment (Sakchai-Refinitiv) [reply .github/workflows/release.yml:48]: @wsuwt this is normal merging with a default message, it should work.

**Review by Sarin-Udompanish — APPROVED** [review]

**Comment by Sakchai-Refinitiv** [comment]
Hi @TremayneChrist, I see your PRs have modified CI jobs, so I do not want to make a conflict with your PR, please review this PR again after you have done. Thanks.

**Comment by TremayneChrist** [comment]
This can be closed, as we're going with manual release trigger on 'default' branch

**Pull request closed without merging**